### DISCUSSION TITLE
Constant anisotropy mobility

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#27789] [Q&A Modules: Phase field]
Hello!
I am working on phase transformation using KKS method. I am trying to implement anisotropic mobility as follows:
```
  [CHBulk]
    type = KKSSplitCHCRes
    variable = c
    ca = cf
    fa_name = ff
    w = w
  []
  [dcdt1]
    type = CoupledTimeDerivative
    variable = w
    v = 'c cb cf eta'
  []
  [ckernel]
    type = SplitCHWResAniso
    variable = w
    mob_name = M
    coupled_variables = 'cb cf eta'
[]
...
[Materials]
  [mob]
    type = ConstantAnisotropicMobility
    tensor = '0.1 0 0
              0   1 0
              0   0 0'
    M_name = M
  []
[]
```
 But I am getting the following error:

```
*** ERROR ***
The following error occurred in the object "mob", of type "ConstantAnisotropicMobility".

The declared non-AD material property 'M' of type 'libMesh::TensorValue<double>'
is already retrieved or declared as a non-AD property of type 'double'.
```

Can you please help me identifying the issue here?
Thank you
Ashish

**Comment by GiudGiud** [comment]
Hello

I think you just need to make that material AD.

Try prefixing the type of the material with AD

**Reply by ashishdhole** [reply]
Hello, 
Can you give me some more information on how to do it? is it something related to MaterialADConverter? Can I find some example for reference?
Thank you
Ashish

**Reply by ashishdhole** [reply]
If I use MaterialADConverter with input and output property name as same 'M' I am getting following error.
```
*** ERROR ***
The following error occurred in the object "non_ad_local_damage", of type "MaterialADConverter".

The requested AD material property 'M' of type 'double'
is already retrieved or declared as a non-AD property of type 'double'.
```


Ashish

**Reply by GiudGiud** [reply]
The MaterialADconverter is not meant for tensors. So you can't use it here

**Reply by ashishdhole** [reply]
What should be the correct way to do it?

**Comment by GiudGiud** [comment]
Try 

Type = ADConstantAnisotropicMobility

**Reply by ashishdhole** [reply]
Still same error,
```
*** ERROR ***
The following error occurred in the object "mob", of type "ADConstantAnisotropicMobility".

The declared AD material property 'M' of type 'libMesh::TensorValue<double>'
is already retrieved or declared as a non-AD property of type 'double'.
```

**Reply by GiudGiud** [reply]
Ok reading the error message again 

You are creating the mobility as a tensor but one of the objects is requesting it as a simple number

**Reply by ashishdhole** [reply]
does KKS CH bulk is causing it? If I run the example script from the website for SplitCHWResAniso it is running, but if I am using same kernels in my script it is not working. is there any other way to define the anisotrpy in kks models?

**Reply by GiudGiud** [reply]
That's likely. You should check the code that kernel to see the type of the material property used in that kernel for mobility

**Reply by ashishdhole** [reply]
I think the material property that is asked by SplitCHWResAniso is tensor (const MaterialProperty<T> & _mob;) yet it still is not accepting the tensor value supplied in the input file.

**Reply by GiudGiud** [reply]
What error are you getting now?

**Reply by ashishdhole** [reply]
Hi, 
Surprisingly it is working now. I made no changes at all. It is strange to me what might have happened before but there is no error now. Although I am not getting the results as expected, but, I think it has something to do with the way I made my input file. So at the moment there is no error. 
Thank you Guillaume.

Ashish
